dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/thread/ScheduledExecutorServiceFactory.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.thread
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.configuration.BackPressureStrategy
+import java.util.concurrent.ScheduledExecutorService
+
+/**
+ * A factory for [ScheduledExecutorService].
+ */
+fun interface ScheduledExecutorServiceFactory {
```

Comment:
it seems this one can be `internal`, if I'm not wrong. It is used only in the `CoreFeature` constructor signature, which is internal.

```suggestion
internal fun interface ScheduledExecutorServiceFactory {
```